manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/AgentHandler.java

```diff
@@ -0,0 +1,314 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.generator.services.plugins;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.models.DecoratedArrayList;
+import io.hyscale.commons.models.ManifestContext;
+import io.hyscale.generator.services.generator.MetadataManifestSnippetGenerator;
+import io.hyscale.generator.services.model.ManifestResource;
+import io.hyscale.generator.services.predicates.ManifestPredicates;
+import io.hyscale.generator.services.utils.ConfigMapDataUtil;
+import io.hyscale.generator.services.utils.PodSpecEnvUtil;
+import io.hyscale.generator.services.utils.SecretsDataUtil;
+import io.hyscale.generator.services.utils.VolumeMountsUtil;
+import io.hyscale.plugin.framework.annotation.ManifestPlugin;
+import io.hyscale.plugin.framework.handler.ManifestHandler;
+import io.hyscale.plugin.framework.models.ManifestSnippet;
+import io.hyscale.plugin.framework.util.JsonSnippetConvertor;
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.*;
+import io.kubernetes.client.models.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+/**
+ * @author nishanth.panthangi created on 05-Nov-2019
+ */
+@Component
+@ManifestPlugin(name = "AgentHandler")
+public class AgentHandler implements ManifestHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(AgentHandler.class);
+    private static final String DEFAULT_IMAGE_PULL_POLICY = "Always";
+    //  private static final String DEFAULT_IMAGE_PULL_POLICY = "IfNotPresent";
+    //  config map name = "agent-{agent.name}-configmap"
+
+    @Override
+    public List<ManifestSnippet> handle(ServiceSpec serviceSpec, ManifestContext manifestContext) throws HyscaleException {
+        if (!ManifestPredicates.getAgentsPredicate().test(serviceSpec)) {
+            logger.debug("Agents found to be empty while processing service spec data.");
+            return null;
+        }
+        TypeReference<List<Agent>> agentsList = new TypeReference<List<Agent>>() {
+        };
+        List<Agent> agents = null;
+        try {
+            agents = serviceSpec.get(HyscaleSpecFields.agents, agentsList);
+        } catch (HyscaleException e) {
+            logger.error("Couldn't fetch agents, returning null",e);
+            return null;
+        }
+        List<ManifestSnippet> manifestSnippetList = new ArrayList<ManifestSnippet>();
+
+        try {
+            manifestSnippetList.addAll(generateConfigMapAndSecretMetadata(agents));
```

Comment:
Agent handler should just be this handle method and any small util methods
All the data building is handled inside the same class which weakens the code readability.
Can you distribute the data building responsibility and construct a single interface to handle them